include/envoy/stream_info/stream_info.h

```diff
@@ -422,13 +422,26 @@ class StreamInfo {
   /**
    * @param connection_info sets the downstream ssl connection.
    */
-  virtual void setDownstreamSslConnection(const Ssl::ConnectionInfo* ssl_connection_info) PURE;
+  virtual void
+  setDownstreamSslConnection(const Ssl::ConnectionInfoConstSharedPtr& ssl_connection_info) PURE;
 
   /**
    * @return the downstream SSL connection. This will be nullptr if the downstream
    * connection does not use SSL.
    */
-  virtual const Ssl::ConnectionInfo* downstreamSslConnection() const PURE;
+  virtual const Ssl::ConnectionInfoConstSharedPtr downstreamSslConnection() const PURE;
```

Comment:
nit: no top level const in return type: https://clang.llvm.org/extra/clang-tidy/checks/readability-const-return-type.html

Reply:
Done